takeDOMSnapshot.js

```diff
@@ -21,25 +40,19 @@ function extractCSSBlocks(doc) {
       const href = element.href || element.getAttribute('href');
       blocks.push({ key: href, href, baseUrl: element.baseURI });
     } else {
-      // <style>
-      const lines = Array.from(element.sheet.cssRules).map((r) => r.cssText);
-
-      // Filter out those lines that are comments (these are often source
-      // mappings)
-      const content = lines.filter((line) => !COMMENT_PATTERN.test(line)).join('\n');
-
+      const content = element.textContent
+        ? stripComments(element.textContent)
+        : getContentFromStyleSheet(element.sheet);
```

Comment:
The naming of these functions seems a bit off to me. Also, I think perhaps it should be refactored a bit where here and below we would call `getContentFromStyleSheet(element)` and then that function would be this:

```js
function getContentFromStyleSheet(element) {
  const lines = element.textContent
    ? element.textContent.split('\n').map(line => line.trim())
    : element.sheet.map((rule) => rule.cssText);

  return lines
    .filter((line) => line && !COMMENT_PATTERN.test(line))
    .join('\n');
}
```

That sorts out the naming and avoids duplicating/forking logic.

Reply:
Oh it looks like adoptedStyleSheets will need slightly different logic.

Reply:
I updated this PR